src/problems/n_puzzle/n_puzzle_problem.py

```diff
@@ -0,0 +1,58 @@
+from base import Problem
+from problems.n_puzzle.n_puzzle_state import NPuzzleState
+from typing import Union, List
+from copy import deepcopy
+
+
+class NPuzzleProblem(Problem):
+    def __init__(self, initial: NPuzzleState, goal: NPuzzleState = None):
+        super().__init__(initial,goal)
+        self.previousStans = []
```

Comment:
1. camelCase 🙀 
2. Nie powinno być `previous_states`? :)

Reply:
To była nie używana zmienna do testowania. Aktualnie jej w ogóle nie ma. :D